### ISSUE TITLE
How to install WSA on Win 11 / Parallels Desktop 17.1.0.51516 / iMac 27 Late 2014 

### ISSUE THREAD

**Issue by dirkseng** [LSPosed/MagiskOnWSA#75]
Recently I successfully installed Win 11 (Enterprise, 21H2 / 22000.258) on my iMac 27 late 2014 (MacOS Big Sur 11.6.1) via Parallels Desktop (Ver. 17.1.0.51516). After then I installed WSA (Windows Subsystem for Android / By MagiskOnWSA) on Windows 11. But it cannot run with the following information. Therefore, a subsequent installation cannot be performed, ie. Google Play Store.

"Ensure that Virtual Machine Platform is enabled in optional Windows features. Ensure that your device has virtualization enabled in the bios. If you're running Windows Subsystem for Android in a VM ensure that you have enabled nested virtualization for the VM on the host.
See http://aka.ms/enablevirtualization."

According to the information prompt, I have conducted the following check:

 1. Checking the Windows Virtual Machine Platform, the Windows Virtual Machine Program Monitor Platform, and the Hyper-V were selected.

 2. Because of iMac 27 doesn't have BIOS, the Virtualization cannot be enabled from UEFI. I try to select the "Recovery --> Start Now --> Troubleshooting --> Advanced Options --> UEFI Fireware Setting, but there is no UEFI at all.

 3. I tried to close the Windows and open the setting of the Control Centre of Parallels Desktop, and to select the CPU/Memory --> Advanced Option under the Hardware Tab: With Apple, the "Nested" is gray, not optional. But the "Nested" under Parallels is optional. After option out of the settings, try to restart Win 11, cannot start. Tip below:

 "Windows 11 cannot be installed because Parallels Hypervisor System Extension is not load." "Tip Action under macOS System Preferences --> Security and Privacy --> General. Click "OK" and agree to load Hypervisor System Extension resolved. But never display the "OK" interface for clicking.

To sum up, I would like to ask the warrior to help solve the following problems:

1.	How to enable Nested Virtualization for Parallels Mac VM (If “Apple”is enabled, the “Nested --> Nested Virtualization” option is gray, and Win 11 cannot be installed or started after selecting Parallels;

2.	How to load the “Parallels Hypervisor System Extension under macOS System Preferences --> Security and Privacy --> General, due to display screen with no pop-up;

3.	How to enter the BIOS / UEFI on the iMac 27 virtual machine, select / Enable Virtualization Technology;

4.  How to create a simulated UEFI startup for the iMac 27 by myself.

5.	Or another command line turns on Nested Virtualization.

 ......

Thanks a lot!

dirkseng / Peking / China
e-mail: dirkseng@126.com

**Comment by okibcn** [comment]
You can't virtualize inside a virtualized Windows 11.  Sell that Mac and get a real PC. You will get more for less.

**Comment by yujincheng08** [comment]
I don't think anyone here could help you with nested virtualization but services from Parallels Desktop.

**Comment by dirkseng** [comment]
Thanks for the tip! In fact, I had left a message on Parallels Desktop before that and got no help!

**Comment by voip2014** [comment]
Its possible to do this on VMware workstation and VMware ESXi, If you use bootcamp you can load the ESXI hypervisor then load Windows 11 via that you can actually run WSA that way if you have an Intel Mac. For ARM based macs your out of luck for WSA at this time. Also any older windows PC with VT extensions can also work for windows 11 with WSA, hyper V seems to work on a lot of harder even older stuff. Great solution for testing andriod apps on windows.

**Comment by dirkseng** [comment]
Thanks for the guidance! 
If I install Windows through BootCamp, It will experience problems with restarting the system. i.e. I have to restart the system when I want to use either Mac or Win. Install Win11 through Parallels Desktop, there is no restart problem, just switch under the same screen! And install WSA, mainly want to install some Google applications, such as: BBC iPlayer, etc.
When it comes to BBC iPlayer app, only the Android system and iPhone are downloaded and installed! There is no “BBC iPlayer” app under Mac and Windows, and can only be installed on Mac or Win via WebCatalog, but this affects the playback speed. So I especially want to own a WSA.
Thanks again.
